pkg/app/api/service/pipedservice/service.proto

```diff
@@ -345,3 +352,21 @@ message ReportApplicationLiveStateEventsRequest {
 message ReportApplicationLiveStateEventsResponse {
     repeated string failed_ids = 1;
 }
+
+message GetLatestEventRequest {
+    string name = 1 [(validate.rules).string.min_len = 1];
+    map<string,string> labels = 2;
+}
+
+message GetLatestEventResponse {
+    pipe.model.Event event = 1 [(validate.rules).message.required = true];
+}
+
+message ListEventsRequest {
+    // Mark that the client has handled all events created before this value.
+    int64 milestone = 1 [(validate.rules).int64.gt = 0];
```

Comment:
nit: This is API definition so I think it could be `from` (if the event for this value should be included too) or `after`.
(Milestone of marked-something is just a name used inside piped code.)

Reply:
Thank you. Acutually I've cared about this part. `from` looks good

Reply:
Exactly we don't include the given value itself, so `after` is more appropriate.